universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramReviewController.java

```diff
@@ -0,0 +1,131 @@
+    return checkApplicantAuthorization(profileUtils, request, applicantId)
+        .thenComposeAsync(
+            v -> applicantService.getReadOnlyApplicantProgramService(applicantId, programId),
+            httpExecutionContext.current())
+        .thenApplyAsync(
+            (roApplicantProgramService) -> {
+              ImmutableList<AnswerData> summaryData = roApplicantProgramService.getSummaryData();
+              // TODO: Get program title. (Currently no way to do that from
+              // roApplicantProgramService)
+              String programTitle = "Program title";
+              return ok(
+                  summaryView.render(
+                      request, applicantId, programId, programTitle, summaryData, banner));
+            },
+            httpExecutionContext.current())
+        .exceptionally(
+            ex -> {
+              if (ex instanceof CompletionException) {
+                Throwable cause = ex.getCause();
+                if (cause instanceof SecurityException) {
+                  return unauthorized();
+                }
+                if (cause instanceof ProgramNotFoundException) {
+                  return notFound(cause.toString());
+                }
+                throw new RuntimeException(cause);
+              }
+              throw new RuntimeException(ex);
+            });
+  }
+
+  @Secure
+  public CompletionStage<Result> submit(Request request, long applicantId, long programId) {
+    return checkApplicantAuthorization(profileUtils, request, applicantId)
+        .thenComposeAsync(
+            v -> {
+              return submit(applicantId, programId);
+            },
+            httpExecutionContext.current())
+        .exceptionally(
+            ex -> {
+              if (ex instanceof CompletionException) {
+                Throwable cause = ex.getCause();
+                if (cause instanceof SecurityException) {
+                  return unauthorized();
+                }
+                if (cause instanceof ProgramNotFoundException) {
+                  return notFound(cause.toString());
+                }
+                throw new RuntimeException(cause);
+              }
+              throw new RuntimeException(ex);
+            });
+  }
+
+  private CompletionStage<Result> submit(long applicantId, long programId) {
+
+    return applicationRepository
+        .submitApplication(applicantId, programId)
+        .thenApplyAsync(
+            applicationMaybe -> {
+              if (applicationMaybe.isEmpty()) {
+                Call reviewPage =
+                    routes.ApplicantProgramReviewController.review(applicantId, programId);
+                return found(reviewPage).flashing("banner", "Error saving application.");
+              }
+              Call endOfProgramSubmission = routes.ApplicantProgramsController.index(applicantId);
+              Application application = applicationMaybe.get();
+              // Placeholder application ID display.
```

Comment:
question: What does this comment mean? In what way is this a placeholder?

Reply:
This was copied over from the stub implementation. 

I suspect that instead of showing
`Successfully saved application: application ID 1` 
we should show
`Successfully submitted application: Application Title`

I'll add this as a cleanup task for #914